src/components/modals/AutoLogoutModal.tsx

```diff
@@ -1,63 +1,59 @@
-import { Appearance, Platform, View } from 'react-native'
+import { Platform, View } from 'react-native'
 import { AirshipBridge } from 'react-native-airship'
-import Feather from 'react-native-vector-icons/Feather'
+import { WheelPicker } from 'react-native-wheel-picker-android'
 
 import { useHandler } from '../../hooks/useHandler'
 import { lstrings } from '../../locales/strings'
 import { DisplayTime, displayToSeconds, secondsToDisplay } from '../../util/displayTime'
-import { ButtonsView } from '../buttons/ButtonsView'
+import { ModalButtons } from '../buttons/ModalButtons'
 import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
-import { EdgeText } from '../themed/EdgeText'
 import { EdgeModal } from './EdgeModal'
 
 interface Props {
   bridge: AirshipBridge<number | undefined>
   autoLogoutTimeInSeconds: number
 }
 
+type UnitOptionValue = 'seconds' | 'minutes' | 'hours' | 'days'
+
 export const AutoLogoutModal = (props: Props) => {
   const { autoLogoutTimeInSeconds, bridge } = props
   const theme = useTheme()
   const styles = getStyles(theme)
 
   const [displayTime, setDisplayTime] = React.useState<DisplayTime>(secondsToDisplay(autoLogoutTimeInSeconds))
 
-  // In Android, the Picker will be hidden with the text overlay.
-  // Determine correct contrasting text color for the native Android Picker
-  // menu. The selected option will show the normal theme color in a text
-  // overlay, but the Picker's color itself must be set to a color that
-  // contrasts with the Android OS theme.
   const isAndroid = Platform.OS === 'android'
-  const textColor = React.useMemo(() => {
-    if (isAndroid) {
-      const colorScheme = Appearance.getColorScheme()
-      return colorScheme === 'dark' ? theme.nativeComponentTextDark : theme.nativeComponentTextLight
-    } else {
-      return theme.pickerText
+
+  const numberLabelsAndroid: string[] = React.useMemo(() => {
+    const out: string[] = []
+    for (let i = 0; i < 60; i++) {
+      out.push(i > 0 ? `${i}` : lstrings.string_disable)
     }
-  }, [theme, isAndroid])
+    return out
+  }, [])
 
-  const numberPickerOptions = [<Picker.Item key="disable" label={lstrings.string_disable} value={0} color={textColor} />]
-  for (let i = 1; i < 60; i++) {
-    const text = String(i)
-    numberPickerOptions.push(<Picker.Item key={text} label={text} value={i} color={textColor} />)
+  const numberOptionsIos: React.ReactElement[] = []
+  for (let i = 0; i < 60; i++) {
+    numberOptionsIos.push(<Picker.Item key={i} label={i === 0 ? lstrings.string_disable : `${i}`} value={i} color={theme.pickerText} />)
   }
 
-  const measurementOptionsStrings = {
-    seconds: lstrings.settings_seconds,
-    minutes: lstrings.settings_minutes,
-    hours: lstrings.settings_hours,
-    days: lstrings.settings_days
-  }
+  /** For linking selected labels to DisplayTime-compatible values */
+  const unitLabelValueList: Array<{ label: string; value: UnitOptionValue }> = React.useMemo(
```

Comment:
Don't use `useMemo`, since it's not saving any performance. Nothing is in JS is faster than raw constants. You could possibly move the constant outside of the render function, at the top-level of the file, to avoid re-evaluating it.

By using `useMemo`, we need to bundle up a callback function and then jump into React code that needs to do a state lookup, and array loop, and so forth. The `useMemo` thing only gives a performance boost if the contents are *really* expensive, like iterating over a giant list of wallets or doing biggystring math.

Reply:
I get a dependency loop for one of the Android datasets, but I'll remove the rest